### DISCUSSION TITLE
TypeError: clip_grad_norm_() got an unexpected keyword argument 'error_if_nonfinite'

### DISCUSSION THREAD

**Discussion by Lizh-TW** [coqui-ai/TTS#746] [General Q&A] [answered]
I'm trying to train a multi-speaker TTS it throws the error.
Please tell me how to solve this problem, thank you!!!

**ERROR**
` ! Run is removed from /workspace/multi-speaker/synthesizer/aishell3-dca-August-18-2021_06+38AM-0000000
Traceback (most recent call last):
  File "/workspace/TTS/TTS/trainer.py", line 848, in fit
    self._fit()
  File "/workspace/TTS/TTS/trainer.py", line 834, in _fit
    self.train_epoch()
  File "/workspace/TTS/TTS/trainer.py", line 686, in train_epoch
    _, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
  File "/workspace/TTS/TTS/trainer.py", line 564, in train_step
    outputs, loss_dict_new, step_time = self._optimize(
  File "/workspace/TTS/TTS/trainer.py", line 494, in _optimize
    grad_norm = torch.nn.utils.clip_grad_norm_(model.parameters(), grad_clip, error_if_nonfinite=False)
TypeError: clip_grad_norm_() got an unexpected keyword argument 'error_if_nonfinite'`

**My config.json**
{
    "datasets": [
        {
            "name": "aishell3",
            "path": "./dataset/aishell_3",
            "meta_file_train": "metadata.csv",
            "meta_file_val": null
        }
    ],
    "audio": {
        "fft_size": 1024,
        "win_length": 1024,
        "hop_length": 256,
        "frame_length_ms": null,
        "frame_shift_ms": null,
        "sample_rate": 22050,
        "preemphasis": 0.0,
        "ref_level_db": 20,
        "do_trim_silence": true,
        "trim_db": 60,
        "power": 1.5,
        "griffin_lim_iters": 60,
        "num_mels": 80,
        "mel_fmin": 50.0,
        "mel_fmax": 7600.0,
        "spec_gain": 1,
        "signal_norm": true,
        "min_level_db": -100,
        "symmetric_norm": true,
        "max_norm": 4.0,
        "clip_norm": true,
        "stats_path": "scale_stats.npy"
    },
    "distributed_backend": "nlcc",
    "distributed_url": "tcp:\/\/localhost:54321",
    "model": "Tacotron2",
    "run_name": "aishell3-dca",
    "run_description": "tacotron2 with dynamic conv attention.",
    "batch_size": 64,
    "eval_batch_size": 16,
    "mixed_precision": true,
    "loss_masking": true,
    "decoder_loss_alpha": 0.25,
    "postnet_loss_alpha": 0.25,
    "postnet_diff_spec_alpha": 0.25,
    "decoder_diff_spec_alpha": 0.25,
    "decoder_ssim_alpha": 0.25,
    "postnet_ssim_alpha": 0.25,
    "ga_alpha": 5.0,
    "stopnet_pos_weight": 15.0,
    "run_eval": true,
    "test_delay_epochs": 10,
    "max_decoder_steps": 1000,
    "noam_schedule": true,
    "grad_clip": 0.05,
    "epochs": 1000,
    "lr": 0.001,
    "wd": 1e-06,
    "warmup_steps": 4000,
    "memory_size": -1,
    "prenet_type": "original",
    "prenet_dropout": true,
    "attention_type": "dynamic_convolution",
    "location_attn": true,
    "attention_norm": "sigmoid",
    "r": 2,
    "stopnet": true,
    "separate_stopnet": true,
    "print_step": 25,
    "tb_plot_step": 100,
    "print_eval": false,
    "save_step": 10000,
    "checkpoint": true,
    "text_cleaner": "phoneme_cleaners",
    "num_loader_workers": 4,
    "num_val_loader_workers": 4,
    "batch_group_size": 4,
    "min_seq_len": 6,
    "max_seq_len": 180,
    "compute_input_seq_cache": true,
    "output_path": "/workspace/multi-speaker/synthesizer/",
    "phoneme_cache_path": "/workspace/multi-speaker/synthesizer/phoneme_cache/",
    "use_phonemes": true,
    "phoneme_language": "zh-CN",
    "use_speaker_embedding": true,
    "use_gst": false,       			
    "use_external_speaker_embedding_file": true,
    "external_speaker_embedding_file": "/workspace/multi-speaker/synthesizer/embedding/speakers.json"
}

**Comment by erogol** [comment, marked as answer]
Try updating your pytorch

**Reply by Lizh-TW** [reply]
Thank you!!!
